pennylane/templates/subroutines/qsvt.py

```diff
@@ -0,0 +1,233 @@
+    """
+    if qml.math.shape(A) == () or qml.math.shape(A) == (1,):
+        A = qml.math.reshape(A, [1, 1])
+
+    c, r = qml.math.shape(A)
+
+    UA = BlockEncode(A, wires=wires, do_queue=False)
+    projectors = []
+
+    if convention == "Wx":
+        angles = _qsp_to_qsvt(angles)
+        global_phase = (len(angles) - 1) % 4
+
+        if global_phase == 1:
+            qml.exp(qml.Identity(wires=wires), 1j * (3 * np.pi / 2))
+
+        elif global_phase == 2:
+            qml.exp(qml.Identity(wires=wires), 1j * np.pi)
+
+        elif global_phase == 3:
+            qml.exp(qml.Identity(wires=wires), 1j * np.pi / 2)
+
+    for idx, phi in enumerate(angles):
+        if idx % 2 == 0:
+            projectors.append(PCPhase(phi, dim=r, wires=wires, do_queue=False))
+        else:
+            projectors.append(PCPhase(phi, dim=c, wires=wires, do_queue=False))
+
+    return QSVT(UA, projectors, wires=wires)
+
+
+class QSVT(Operation):
+    r"""Implements the
+    `quantum singular value transformation <https://arxiv.org/abs/1806.01838>`__ circuit.
+
+    Given a circuit :math:`U(A)`, which block encodes the matrix :math:`A`, and a list of projector-controlled
+    phase shifts, this template applies the circuit for quantum singular value transformation.
+
+    .. math::
+
+        \begin{align}
+             U_{QSVT}(A, \phi) &=
+             \begin{bmatrix}
+                Poly^{SV}(A) & \cdot \\
+                \cdot & \cdot
+            \end{bmatrix}.
+        \end{align}
+
+    This circuit can be used to perform the standard quantum singular value transformation algorithm, consisting
+    of alternating block encoding and controlled phase shift operations.
+
+    Args:
+        UA (Operator): the block encoding circuit, specified as an :class:`~.Operator`
+            or quantum function.
+        projectors (Sequence[Operator]): a list of projector-controlled phase
+            shifts that implement the desired polynomial.
+        wires (Iterable): the wires the template acts on.
+
+    Raises:
+        ValueError: if the input block encoding is not an operator.
+
+    .. details::
+        :title: Usage Details
+
+        .. code-block:: python
+
+            dev = qml.device("default.qubit", wires=2)
+            A = [[0.1]]
+            blckencode = qml.BlockEncode(A, wires=[0, 1])
+            lst_phis = [qml.PCPhase(i + 0.1, dim=1, wires=[0, 1]) for i in range(3)]
+
+            @qml.qnode(dev)
+            def example_circuit(A):
+                qml.QSVT(blckencode, lst_phis, wires=[0, 1])
+                return qml.expval(qml.PauliZ(wires=0))
+
+            print(qml.matrix(example_circuit)(A))
+
+            [[-0.11-0.01j -0.58+0.8j   0.  +0.j    0.  +0.j  ]
+             [ 0.45+0.89j -0.11-0.01j  0.  +0.j    0.  +0.j  ]
+             [ 0.  +0.j    0.  +0.j    1.  +0.j    0.  +0.j  ]
+             [ 0.  +0.j    0.  +0.j    0.  +0.j    1.  +0.j  ]]
```

Comment:
```suggestion
             [ 0.  +0.j    0.  +0.j    0.  +0.j    1.  +0.j  ]])
```

Reply:
No parenthesis required if it is not showing "array(" in the beginning